access/slack/Dockerfile

```diff
@@ -0,0 +1,23 @@
+# Build the plugin binary
+FROM golang:1.16.9 as builder
+
+WORKDIR /workspace
+
+# Copy the Go Modules manifests
+COPY go.mod go.mod
+COPY go.sum go.sum
+
+# Copy the go source
+COPY access/slack access/slack
+COPY vendor vendor
+COPY lib lib
+
+# Build
+RUN make -C access/slack
+
+# Use distroless as minimal base image to package the manager binary
+# Refer to https://github.com/GoogleContainerTools/distroless for more details
+FROM gcr.io/distroless/base:latest
```

Comment:
I'm not very knowledgeable in the area, but distroless does seem nice for the plugins. However, the previous [dockerize slack PR](https://github.com/gravitational/teleport-plugins/pull/199) had some comments about copying aspects of the teleport dockerfile, such as using Ubuntu 18.04 or 20.04 as a base.  Has this already been internally discussed, and do we plan on making other plugin docker images using the base image?

Reply:
@Joerger 

Nope it wasn't discussed. This PR is a request for such a discussion :)

Initially, the idea of using distroless came up when I was playing with Kubebuilder framework while working on #319. The framework generated a `Dockerfile` that used `distroless:static` and I was really impressed by the resulting image size.

Then I decided to try it for Slack plugin and it failed. Then I tried `distroless:base` (it's the same as `distroless:static` but contains glibc) and it was okay. It turned out that `distroless:static` image is for static Go binaries built without cgo.

`CGO_ENABLED=1" is another thing that we copied from Teleport's `Makefile` and I don't understand why :) So maybe it's time to stop copying things from Teleport blindly? I don't see an argument why we might need Ubuntu for such a small things like plugins.

Want to invite @russjones and @webvictim for the discussion.

Reply:
We should probably bin the base image to a specific version (or hash) as well. We should opt-in to updates, rather than be surprised by them ;-)

Reply:
Yes but distroless images don't have a versioning scheme I suppose. So I just used the latest hash